Assets/Plugins/README.md

```diff
@@ -108,41 +104,135 @@ public async void OnSendTransactionClicked()
+    _responseId = SM.MavisId.OnSignTypeData(typedData);
+    string responseData = await WaitForMavisIdResponse(_responseId);
+    Debug.Log(responseData);
+}
+```
+
+### Call contracts
+
+Allows another contract to spend 1 AXS on user's behalf, returning a transaction hash.
 
 ```csharp
 public async void OnApproveErc20Clicked()
+{
+    // Address of the smart contract to interact with
+    string contractAddress = "0x3c4e17b9056272ce1b49f6900d8cfd6171a1869d";
+
+    string readableAbi = "function approve(address _spender, uint256 _value)";
+    // Approve 1 AXS
+    var approveParams = new { _spender = "0x6B190089ed7F75Fe17B3b0A17F6ebd69f72c3F63", _value = 1000000000000000000 };
+    try
     {
-        string contractAddress = "0x3c4e17b9056272ce1b49f6900d8cfd6171a1869d";
-
-        string readableAbi = "function approve(address _spender, uint256 _value)";
-        // Approve 1 AXS
-        var approveParams = new { _spender = "0x6B190089ed7F75Fe17B3b0A17F6ebd69f72c3F63", _value = 1000000000000000000 };
-        try
-        {
-            var data = ABI.EncodeFunctionData(readableAbi, approveParams);
-            Debug.Log("Approve data : " + data);
-            _responseId = SM.MavisId.OnCallContract(contractAddress, data);
-            string responseData = await WaitForMavisIdResponse(_responseId);
-            Debug.Log(responseData);
-
-        }
-        catch (System.Exception e)
-        {
-            Debug.Log("Error in call contract : " + e.Message);
-        }
+        // Transaction data to be sent to the smart contract
+        var data = ABI.EncodeFunctionData(readableAbi, approveParams);
+        Debug.Log("Approve data: " + data);
+        _responseId = SM.MavisId.OnCallContract(contractAddress, data);
+        string responseData = await WaitForMavisIdResponse(_responseId);
+        Debug.Log(responseData);
 
     }
+    catch (System.Exception e)
+    {
+        Debug.Log("Error in call contract: " + e.Message);
+    }
+}
+```
+
+## Utilities
+
+The SM.ID.Utils namespace provides utility functions for easier wallet interactions. The following are some common tasks.
+
+### Encode function data
+
+Encodes function data for smart contract interactions.
+
+```csharp
+string EncodeFunctionData(string ABI, object values)
```

Comment:
Please note that, this is readable ABI, that mean the function signature of contract as example , not the json ABI.

Reply:
Can you suggest the right way to phrase it?

Reply:
It's smart contract function like : function approve(address _spender, uint256 _value) which you can see clearly in smart contract 
For example : https://github.com/OpenZeppelin/openzeppelin-contracts/blob/c304b6710b4b5fcf2a319ad28c36c49df6caef14/contracts/token/ERC20/ERC20.sol#L127